Source/WebKit/UIProcess/XR/ios/WKARPresentationSession.mm

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#import "WKARPresentationSession.h"
+
+#if ENABLE(WEBXR) && USE(ARKITXR_IOS)
+
+#import <Metal/Metal.h>
+#import <wtf/WeakObjCPtr.h>
+
+#undef USE_ARPRESENTATIONSESSION_NOISY_LOG
+#if USE(ARPRESENTATIONSESSION_NOISY_LOG)
+#define NOISY_RELEASE_LOG(channel, ...) RELEASE_LOG(channel, __VA_ARGS__)
+#else
+#define NOISY_RELEASE_LOG(channel, ...) ((void)0)
+#endif
+
+NS_ASSUME_NONNULL_BEGIN
+
+#pragma mark - WKARPresentationSessionDescriptor
+
+@implementation WKARPresentationSessionDescriptor {
+    WeakObjCPtr<UIViewController> _presentingViewController;
+}
+
+- (instancetype)copyWithZone:(nullable NSZone *)zone
+{
+    WKARPresentationSessionDescriptor* descriptor = [[[self class] allocWithZone:zone] init];
+    descriptor.presentingViewController = self.presentingViewController;
+
+    return descriptor;
+}
+
+- (nullable UIViewController*)presentingViewController
```

Comment:
No `nullable` here either.

Reply:
And other places.

Reply:
Compiler disagrees: `Nullability specifier 'nonnull' conflicts with existing specifier '_Nullable'`